cli/internal/ui/ui.go

```diff
@@ -15,8 +15,11 @@ import (
 
 const ansiEscapeStr = "[\u001B\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;[a-zA-Z\\d]*)*)?\u0007)|(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PRZcf-ntqry=><~]))"
 
+// IsTTY is true when stdout appears to be a tty
 var IsTTY = isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd())
-var IsCI = os.Getenv("CI") == "true" || os.Getenv("BUILD_NUMBER") == "true" || os.Getenv("TEAMCITY_VERSION") != ""
+
+// IsCI is true when we appear to be running in a non-interactive context.
+var IsCI = !IsTTY || os.Getenv("CI") != "" || os.Getenv("BUILD_NUMBER") != "" || os.Getenv("TEAMCITY_VERSION") != ""
```

Comment:
How about split checks?

- `IsInteractive` (the thing we care about needs user input)
- `IsPersistent`(the thing we care about is whether or not a long-running task is viable, or if after I yield the whole thing is getting canned)
- `IsCI` (explicitly detect a CI environment, if for some reason the other two weren't good enough)

Reply:
I think that makes sense if we end up needing to make a more distinctions. This is the second use of this variable. If we end up needing to tweak it, I agree, going by feature makes sense.